src/snowflake/snowpark/modin/plugin/_internal/ordered_dataframe.py

```diff
@@ -1052,6 +1052,7 @@ def join(
         right: "OrderedDataFrame",
         left_on_cols: Optional[list[str]] = None,
         right_on_cols: Optional[list[str]] = None,
+        on_comparators: Optional[list["MatchComparator"]] = None,  # type: ignore[name-defined]  # noqa: F821
```

Comment:
@sfc-gh-nkrishna the fact we only support equal null here is mainly for performance purpose, for your usage, will they be able to be converted to equal null, or use asof?